### PR TITLE
Deprecate `update_attributes` and `update_attributes!`

### PR DESCRIPTION
**Pull request by elebow**
`update_attributes` and `update_attributes!` have been aliases for `update` and `update!` since https://github.com/rails/rails/pull/8705 in 2013. That PR says it was a "soft deprecation" and that the longer method names would be removed in the future.

This would be a desirable change because:
  1. It makes application code easier to read by reducing confusion with `update_attribute`/`update_attribute!`, and
  1. The more direct names help new users understand that these are the "normal" ways to modify attributes of an Active Record object.

Is this still the intention? `update_attributes` is still used in a few tests and one place in `ActiveRecord::InternalMetadata`, and mentioned in the AR Callbacks guide; I can change them to the newer forms in this PR if that is the goal.

### CODE REVIEW

**Comment by sikachu** [comment]
5 years ago, wow ... 

r? @guilleiguaran

**Comment by rafaelfranca** [comment]
Yeah, I think it is time to deprecate them. Can you remove all the usages and add tests to check if those methods are deprecated?

**Review by guilleiguaran — CHANGES_REQUESTED** [review]
Please replace all the usages of `update_attributes` with `update` in the test suite

File: activerecord/test/cases/persistence_test.rb [hunk 1]
```diff
@@ -934,42 +934,33 @@ def test_update
...
-      topic.update_attributes(id: 3, title: "Hm is it possible?")
+      topic.update(id: 3, title: "Hm is it possible?")
     end
     assert_not_nil error.cause
     assert_not_equal "Hm is it possible?", Topic.find(3).title
 
-    topic.update_attributes(id: 1234)
+    topic.update(id: 1234)
     assert_nothing_raised { topic.reload }
     assert_equal topic.title, Topic.find(1234).title
```
Comment (elebow) [inline activerecord/test/cases/persistence_test.rb]: Some of these cases were present in `test_update_attributes` but not in `test_update`. They are now moved to `test_update`.

**Review by guilleiguaran — APPROVED** [review]

**Comment by jeremy** [comment]
Thanks @elebow!

**Comment by sgrif** [comment]
Are these aliases causing some sort of maintenance burden? If we actually fully deprecate/remove these methods we are breaking a *lot* of code for virtually no reason. I'm all for keeping them soft-deprecated, but if they aren't causing a maintenance burden I see no reason to actually remove them.

**Comment by pat** [comment]
Just chiming in to raise the same concerns as @sgrif - I think removing these methods could break a significant amount of code in so many gems, and that feels unwise to me.

I personally didn't know `update` was actually the preferred style until maybe early 2018 - there'd never been any warnings about `update_attributes`, so I know I've taken the 'outdated' approach for quite a long time, and I'm sure plenty of other gem developers have as well.

**Pull request closed without merging**